### PR TITLE
Clean up error and logging output

### PR DESCRIPTION
**Pull request by davis-gomes**

### CODE REVIEW

File: uninstall/uninstall-steps/0-uninstall-applications.sh [hunk 1]
```diff
@@ -13,13 +13,13 @@ set -o pipefail
 function initializing_uninstall {
   # Deleting rancher through API
   log "Deleting Rancher through API"
-  rancher_host_name="$(kubectl get ingress -n cattle-system --no-headers -o custom-columns=":spec.rules[0].host")" || return $?
```
Comment (desagar) [inline uninstall/uninstall-steps/0-uninstall-applications.sh:16]: are you sure the semicolon will be parsed by Mac and all Unix flavors as `(kubectl command) || (print error; return $?)`, and not `(kubectl command || print error) ; return $?`

Comment (davis-gomes) [reply uninstall/uninstall-steps/0-uninstall-applications.sh:16]: I am not 100% sure. Do you know a way to ensure that this happens?

Comment (desagar) [reply uninstall/uninstall-steps/0-uninstall-applications.sh:16]: On my Mac, I see this behavior - the second case is problematic and indicates the stuff after the semicolon will happen even if the command succeeds.

```
First Command does not succeed:

ls hello || echo one;echo two
ls: hello: No such file or directory
one
two
```

```
First command succeeds: 
ls server.js || echo one;echo two

server.js
two
```

Comment (desagar) [reply uninstall/uninstall-steps/0-uninstall-applications.sh:16]: parentheses around the stuff to the right of the `||` seems to work but I am not sure if it's ok to use. Needs a bit of research into how Unix treats parens (I know in some cases it is treated as a subshell)

Comment (davis-gomes) [reply uninstall/uninstall-steps/0-uninstall-applications.sh:16]: Okay, thanks. I will look into this.

Comment (desagar) [reply uninstall/uninstall-steps/0-uninstall-applications.sh:16]: Also, I just noticed you are returning $?, which will be the exit status of the `error` function call, not the original kubectl command

Comment (davis-gomes) [reply uninstall/uninstall-steps/0-uninstall-applications.sh:19]: added a function to fix these errors.

File: uninstall/uninstall-steps/0-uninstall-applications.sh [hunk 2]
```diff
@@ -46,17 +46,19 @@ function initializing_uninstall {
 }
 
 function delete_bindings {
+  log "Deleting VerrazzanoBindings"
   kubectl get crd verrazzanobindings.verrazzano.io || return 0
   kubectl get VerrazzanoBindings --no-headers -o custom-columns=":metadata.name" \
     | xargsr kubectl delete VerrazzanoBindings \
```
Comment (desagar) [inline uninstall/uninstall-steps/0-uninstall-applications.sh:52]: is there an xargsr command or is this a typo?

Comment (davis-gomes) [reply uninstall/uninstall-steps/0-uninstall-applications.sh:52]: There is a new command from another PR to handle how xargs works on Mac vs Linux.

File: uninstall/uninstall-steps/1-uninstall-istio.sh [hunk 3]
```diff
@@ -20,73 +20,75 @@ CONFIG_DIR=$INSTALL_DIR/config
...
+    || err_return $? "Could not delete ClusterRoles from Istio" || return $? # return on pipefail
 }
 
 function delete_secrets() {
   # Delete istio.default in all namespaces
-  log "Collecting istio secrets for deletion"
-  kubectl delete secret istio.default --ignore-not-found=true || return $?
-  kubectl delete secret istio.default -n kube-public --ignore-not-found=true || return $?
-  kubectl delete secret istio.default -n kube-node-lease --ignore-not-found=true || return $?
+  log "Retrieveing istio secrets for deletion"
```
Comment (desagar) [inline uninstall/uninstall-steps/1-uninstall-istio.sh:57]: Spelling :-) 
"Retrieving"

File: uninstall/uninstall-steps/1-uninstall-istio.sh [hunk 4]
```diff
@@ -20,73 +20,75 @@ CONFIG_DIR=$INSTALL_DIR/config
...
+  kubectl delete secret istio.default --ignore-not-found=true || err_return $? "Could not delete secret from Istio in namespace default" || return $?
+  kubectl delete secret istio.default -n kube-public --ignore-not-found=true || err_return $? "Could not delete secret from Istio in namespace kube-public" || return $?
+  kubectl delete secret istio.default -n kube-node-lease --ignore-not-found=true || err_return $? "Could not delete secret from Istio in namespace kuce-node-lease" || return $?
```
Comment (desagar) [inline uninstall/uninstall-steps/1-uninstall-istio.sh:60]: typo => message says "kuce-node-lease" instead of "kube-node-lease"

Comment (davis-gomes) [reply uninstall/uninstall-steps/1-uninstall-istio.sh]: This has been fixed.

File: uninstall/uninstall-steps/1-uninstall-istio.sh (hunk 3 above)
Comment (davis-gomes) [reply uninstall/uninstall-steps/1-uninstall-istio.sh]: Find and replace failed me! I just updated this.

**Review by desagar — DISMISSED** [review]

**Review by desagar — APPROVED** [review]

**Pull request merged**